test/performance/event_handler_test.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * This script simulates the creation of new pods in the cluster and measures
+ * the number of events processed by YHS.
+ * It calculates and logs the total number of pods created
+ * and the total number of events processed by YHS during the test.
+ */
+import { Kubernetes } from 'k6/x/kubernetes';
+import { sleep } from 'k6';
+import http from 'k6/http';
+
+const namespace = __ENV.NAMESPACE || 'default';
+const yhsUrl = "http://localhost:8989";
+
+const podTemplate = {
+    apiVersion: 'v1',
+    kind: 'Pod',
+    metadata: {
+        generateName: 'test-pod-',
+        labels: {
+            app: 'performance-by-k6'
+        },
+        namespace: namespace
+    },
+    spec: {
+        containers: [
+            {
+                name: 'test-container',
+                image: 'nginx:latest',
+                ports: [{ containerPort: 80 }]
+            }
+        ]
+    }
+};
+
+export const options = {
+    stages: [
+        { duration: '5s', target: 5 },
+        { duration: '5s', target: 5 },
+        { duration: '5s', target: 0 },
+    ],
+};
+
+const getStat = () => {
+    const res = http.get(`${yhsUrl}/ws/v1/event-statistics`);
+    return JSON.parse(res.body);
+};
+
+export default function () {
+    const kubernetes = new Kubernetes();
+    try {
+        kubernetes.create(podTemplate);
+    } catch (error) {
+        console.log(`Failed to create pod: ${error}`);
+    }
+}
+
+export function teardown(data) {
+    // It takes some time for the events to be processed by YHS
+    sleep(5);
+    const json = getStat();
+    const eventProcessedByYHS = json["APP-ADD"]
+    const kubernetes = new Kubernetes();
+    const pods = kubernetes.list("Pod", namespace);
+    console.log(`Number of NEW pods submitted: ${pods.length}`);
+    console.log(`Number of NEW Event processed by YHS: ${eventProcessedByYHS}`);
```

Comment:
Can you configure `k6` to write a JSON report with some metrics and stats? There should be in-built support

Reply:
A performance test should generate a report with stats and metrics